strconv/traits.mbt

```diff
@@ -13,17 +13,21 @@
 // limitations under the License.
 
 pub trait FromStr {
-  from_string(String) -> Result[Self, String]
+  from_string(String) -> Self!String
 }
 
-pub fn FromStr::from_string(str : String) -> Result[Bool, String] {
-  parse_bool(str)
+impl FromStr for Bool with from_string(str){
+  parse_bool(str)!
 }
```

Comment:
TODO: maybe we should just call it Bool::from_string instead of parse_bool?

Reply:
From a user's POV, I think `parse_*` is more intuitive and easy to find?

<img width="262" alt="image" src="https://github.com/moonbitlang/core/assets/29861494/8d348795-6293-4e21-ac8c-393fecf6dd67">